### PR TITLE
Disable warnings for AMQP tests

### PR DESCRIPTION
**Pull request by LarryOsterman**
# Disable MSVC warnings for AMQP tests in nightly builds

This mirrors the changes made to all the other test components.

Also added cmakepreset values for Linux builds - gcc-9 is the only flavor supported for now, because that is the one which was failing in the CI pipelines. Moved the cmake preset schema to version 3, which enables conditional inclusion (requires cmake 22).

## Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Comment by LarryOsterman** [comment]
Ping - this blocks the nightly tests for C++.

File: sdk/core/azure-core-amqp/test/ut/CMakeLists.txt [hunk 1]
```diff
@@ -31,6 +31,22 @@ add_executable(azure-core-amqp-tests
...
+  #             - 'testing::internal::Mutex::critical_section_init_phase_'
+  #             - 'testing::internal::Mutex::owner_thread_id_'
+  #             - 'testing::internal::Mutex::type_'
+  #           is uninitialized. Always initialize member variables (type.6).
+  # - C26812: The enum type
+  #             - 'testing::internal::Mutex::StaticConstructorSelector'
+  #             - 'testing::TestPartResult::Type'
+  #           is unscoped. Prefer 'enum class' over 'enum' (Enum.3)
+  # - C6326: Google comparisons
+  target_compile_options(azure-core-amqp-tests PUBLIC /wd26495 /wd26812 /wd6326 /wd28204 /wd28020 /wd6330 /wd4389)
```
Comment (ahsonkhan) [inline sdk/core/azure-core-amqp/test/ut/CMakeLists.txt:47]: These are similar to https://github.com/Azure/azure-sdk-for-cpp/blob/5a348b6766cc58903a3c8ee2effc841ae06194c9/sdk/core/azure-core/test/ut/CMakeLists.txt#L107

**Review by ahsonkhan — APPROVED** [review]

File: sdk/core/azure-core-amqp/test/ut/CMakeLists.txt (hunk 1 above)
Comment (LarryOsterman) [reply sdk/core/azure-core-amqp/test/ut/CMakeLists.txt:47]: It should be identical (they are cut and pasted).

What is different?

Comment (ahsonkhan) [reply sdk/core/azure-core-amqp/test/ut/CMakeLists.txt:47]: It **is** identical.

**Pull request merged**